kubernetes-client-api/src/main/java/io/fabric8/kubernetes/client/Config.java

```diff
@@ -1686,11 +1747,77 @@ public void setCurrentContext(NamedContext context) {
   /**
    *
    * Returns the path to the file that this configuration was loaded from. Returns {@code null} if no file was used.
+   * 
+   * @deprecated use {@link #getFiles} instead.
    *
-   * @return the path to the kubeConfig file
+   * @return the kubeConfig file
    */
   public File getFile() {
-    return file;
+    if (Utils.isNotNullOrEmpty(files)) {
+      return files.get(0);
+    } else {
+      return null;
+    }
+  }
+
+  /**
+   * Returns the kube config files that are used to configure this client.
+   * Returns the files that are listed in the KUBERNETES_KUBECONFIG_FILES env or system variables.
+   * Returns the default kube config file if it's not set'.
+   *
+   * @return
+   */
+  public List<File> getFiles() {
+    return files;
+  }
+
+  public KubeConfigFile getFileWithAuthInfo(String name) {
+    if (Utils.isNullOrEmpty(name)
+        || Utils.isNullOrEmpty(getFiles())) {
+      return null;
+    }
+    return getFiles().stream()
+        .filter(Config::isReadableKubeconfFile)
+        .map(file -> {
+          try {
+            return new KubeConfigFile(file, KubeConfigUtils.parseConfig(file));
+          } catch (IOException e) {
+            return null;
+          }
+        })
+        .filter(Objects::nonNull)
+        .filter(entry -> hasAuthInfoNamed(name, entry.getConfig()))
+        .findFirst()
+        .orElse(null);
+  }
+
+  private boolean hasAuthInfoNamed(String username, io.fabric8.kubernetes.api.model.Config kubeConfig) {
+    if (Utils.isNullOrEmpty(username)
+        || kubeConfig == null
+        || kubeConfig.getUsers() == null) {
+      return false;
+    }
+    return kubeConfig.getUsers().stream()
+        .anyMatch(namedAuthInfo -> username.equals(namedAuthInfo.getName()));
+  }
+
+  public static class KubeConfigFile {
```

Comment:
Do we want to have this file in this class?

Reply:
@rohanKanojia we can move it out if you prefer.